pkg/signature/kms/azure/signer.go

```diff
@@ -70,11 +70,6 @@ func LoadSignerVerifier(defaultCtx context.Context, referenceStr string) (*Signe
 		return nil, err
 	}
 
-	a.hashFunc, _, err = a.client.getKeyVaultHashFunc(defaultCtx)
```

Comment:
This was causing the bug reported in https://github.com/sigstore/cosign/issues/3094. The call to `getKeyVaultHashFunc` would fail because it would try to get the hash function of a non-existent key.